internal/infraestructure/github-connection/github-connection.go

```diff
@@ -2,54 +2,185 @@ package github_connection
+		}
+		for _, file := range files {
+			if file.GetAdditions() == 0 {
+				continue
+			}
+			allChanges = append(allChanges, *file)
+		}
+
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
+	}
+
+	return allChanges, nil
 }
 
-func (receiver GithubConnection) CreateComment(comment string) error {
+func (receiver *GithubConnection) CreateComment(files []github.CommitFile) error {
 	ctx := context.Background()
 
-	comment_text := &github.IssueComment{
-		Body: github.String(comment),
+	for _, file := range files {
+		fmt.Printf("Filename: %s\n", file.GetFilename())
+		fmt.Printf("Patch: %s\n", file.GetPatch())
+
+		comments, err := analyzeFileAndCreateComments(&file)
+		if err != nil {
+			return fmt.Errorf("error analyzing file %s: %w", file.GetFilename(), err)
+		}
+
+		for _, commentData := range comments {
+			comment, _, err := receiver.Client.PullRequests.CreateComment(
+				ctx,
+				receiver.RepoOwner,
+				receiver.RepositoryName,
+				receiver.PullRequestNumber,
+				commentData)
+			if err != nil {
+				return fmt.Errorf("error creating PullRequest comment: %w", err)
+			}
+			fmt.Printf("Review comment created: %s\n", comment.GetHTMLURL())
+		}
 	}
 
-	_, _, err := receiver.client.Issues.CreateComment(
-		ctx, receiver.RepoOwner,
-		receiver.RepositoryName,
-		receiver.PullRequestNumber,
-		comment_text)
-	return err
+	return nil
+}
+
+func analyzeFileAndCreateComments(file *github.CommitFile) ([]*github.PullRequestComment, error) {
+	var comments []*github.PullRequestComment
+	patch := file.GetPatch()
+	lines := strings.Split(patch, "\n")
+
+	position := 0
+	newLineNumber := 0
+	startLine := 0
+	endLine := 0
+	var newLines []string
+
+	for _, line := range lines {
+		position++
+		if strings.HasPrefix(line, "+") {
+			newLineNumber++
+			if startLine == 0 {
+				startLine = newLineNumber
+			}
+			endLine = newLineNumber
+			newLines = append(newLines, strings.TrimPrefix(line, "+"))
+		} else {
+			if startLine != 0 {
+				comment := createCommentForLines(file, position-1, startLine, endLine, newLines)
+				comments = append(comments, comment)
+				startLine = 0
+				endLine = 0
+				newLines = nil
+			}
+		}
+	}
+
+	// Handle case where file ends with new lines
+	if startLine != 0 {
+		comment := createCommentForLines(file, position, startLine, endLine, newLines)
+		comments = append(comments, comment)
+	}
+
+	return comments, nil
+}
+
+func createCommentForLines(file *github.CommitFile, position, startLine, endLine int, lines []string) *github.PullRequestComment {
+	commentBody := fmt.Sprintf("Code Review %s - New lines %d to %d:\n\n", time.Now().Format("2006-01-02 15:04:05"), startLine, endLine)
+	commentBody += strings.Join(lines, "\n")
+
+	return &github.PullRequestComment{
+		Body:     github.String(commentBody),
+		CommitID: github.String(os.Getenv("GITHUB_SHA")),
+		Path:     github.String(file.GetFilename()),
+		Position: github.Int(position),
+	}
 }
```

Comment:
Code Review 2024-08-18 02:28:52 - New lines 107 to 159:

	return nil
}

func analyzeFileAndCreateComments(file *github.CommitFile) ([]*github.PullRequestComment, error) {
	var comments []*github.PullRequestComment
	patch := file.GetPatch()
	lines := strings.Split(patch, "\n")

	position := 0
	newLineNumber := 0
	startLine := 0
	endLine := 0
	var newLines []string

	for _, line := range lines {
		position++
		if strings.HasPrefix(line, "+") {
			newLineNumber++
			if startLine == 0 {
				startLine = newLineNumber
			}
			endLine = newLineNumber
			newLines = append(newLines, strings.TrimPrefix(line, "+"))
		} else {
			if startLine != 0 {
				comment := createCommentForLines(file, position-1, startLine, endLine, newLines)
				comments = append(comments, comment)
				startLine = 0
				endLine = 0
				newLines = nil
			}
		}
	}

	// Handle case where file ends with new lines
	if startLine != 0 {
		comment := createCommentForLines(file, position, startLine, endLine, newLines)
		comments = append(comments, comment)
	}

	return comments, nil
}

func createCommentForLines(file *github.CommitFile, position, startLine, endLine int, lines []string) *github.PullRequestComment {
	commentBody := fmt.Sprintf("Code Review %s - New lines %d to %d:\n\n", time.Now().Format("2006-01-02 15:04:05"), startLine, endLine)
	commentBody += strings.Join(lines, "\n")

	return &github.PullRequestComment{
		Body:     github.String(commentBody),
		CommitID: github.String(os.Getenv("GITHUB_SHA")),
		Path:     github.String(file.GetFilename()),
		Position: github.Int(position),
	}